packages/miniflare/src/workers/kv/constants.ts

```diff
@@ -116,3 +117,38 @@ export function testSiteRegExps(
 	if (regExps.exclude !== undefined) return !testRegExps(regExps.exclude, key);
 	return true;
 }
+
+// Magic prefix: if a URLs pathname starts with this, it shouldn't be cached.
+// This ensures edge caching of Workers Assets files is disabled, and the latest
+// local version is always served.
+export const ASSETS_NO_CACHE_PREFIX = "$__MINIFLARE_ASSETS__$/";
+
+export function getAssetsBindingsNames(
+	// __STATIC_CONTENT and __STATIC_CONTENT_MANIFEST binding names are
+	// reserved for Workers Sites. Since we want to allow both sites and
+	// assets to work side by side, we cannot use the same binding name
+	// for assets. Therefore deferring to a different default naming here.
+	assetsKVBindingName = "__STATIC_ASSETS_CONTENT",
+	assetsManifestBindingName = "__STATIC_ASSETS_CONTENT_MANIFEST"
+) {
+	return {
+		ASSETS_KV_NAMESPACE: assetsKVBindingName,
+		ASSETS_MANIFEST: assetsManifestBindingName,
+	} as const;
+}
+
+export function encodeAssetsKey(key: string): string {
+	// `encodeURIComponent()` ensures `ETag`s used by `@cloudflare/kv-asset-handler`
+	// are always byte strings.
+	return ASSETS_NO_CACHE_PREFIX + encodeURIComponent(key);
+}
+export function decodeAssetsKey(key: string): string {
+	return key.startsWith(ASSETS_NO_CACHE_PREFIX)
+		? decodeURIComponent(key.substring(ASSETS_NO_CACHE_PREFIX.length))
+		: key;
+}
+
+export function isAssetsRequest(request: { url: string }) {
+	const url = new URL(request.url);
+	return url.pathname.startsWith(`/${ASSETS_NO_CACHE_PREFIX}`);
+}
```

Comment:
indeed. I left it in because I was busy last night digging into the whole caching situation, trying to understand how it works. I was hoping there's a quick solution to it all, but doesn't seem like that.

If I'm understanding the code correctly, it would seem that while Miniflare does provide us with a caching primitive via the Cache Plugin, `PUT`ing/`GET`ing from cache is not done automatically for us? I guess that was my expectation, that by "just" having the Cache plugin in place, requests would automatically be intercepted and paths added to or read from cache. But for example, in the case of Workers Sites, it would seem that it's the `kv-assets-handler` that's responsible for handling caching, both adding to cache [here](https://github.com/cloudflare/workers-sdk/blob/main/packages/kv-asset-handler/src/index.ts#L307-L320) and reading from cache [here](https://github.com/cloudflare/workers-sdk/blob/main/packages/kv-asset-handler/src/index.ts#L263-L265).

So by the loos of 👆  I'd gather we'd need a similar mechanism in place, where the actual cache calls are done in ASW2? Unless I'm misunderstanding how this works :/

Reply:
will go ahead and remove any assets caching things out of this PR and will track caching specifically in WC-2508